### PR TITLE
ConsoleInteraction.py: Show file path of bear

### PR DESCRIPTION
**Pull request by yash-nisar**
The absolute path of the bear will be shown
with the show-details argument.

Closes https://github.com/coala/coala/issues/2663

### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [x] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

File: coalib/bears/Bear.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import traceback
+import inspect
```
Comment (satwikkansal) [inline coalib/bears/Bear.py]: Imports should be in alphabetical order

File: coalib/bears/Bear.py [hunk 2]
```diff
@@ -155,6 +156,13 @@ def can_detect(cls):
         return cls.CAN_DETECT | cls.CAN_FIX
 
     @classproperty
+    def file_path(cls):
+        """
+        :return: The file where the bear was fetched from.
```
Comment (satwikkansal) [inline coalib/bears/Bear.py]: I think it should be `The file path where the bear was fetched from`

File: coalib/bears/Bear.py [hunk 3]
```diff
@@ -155,6 +156,13 @@ def can_detect(cls):
...
+        """
+        return inspect.getfile(cls)
```
Comment (satwikkansal) [inline coalib/bears/Bear.py]: IMO we should use `inspect.getsourcefile`

Comment (yash-nisar) [inline coalib/bears/Bear.py]: Since all bears are defined in python source files, both will point to the same path.

**Comment by satwikkansal** [comment]
ack 0bd93d5

File: coalib/output/ConsoleInteraction.py [hunk 4]
```diff
@@ -749,6 +749,8 @@ def show_bear(bear,
                          '  ',
                          'This bear cannot fix issues or does not provide '
                          'information about what categories it can fix.')
+        console_printer.print(
+            '  Path:\n' + '   ' + bear.file_path + '\n\n')
```
Comment (adtac) [inline coalib/output/ConsoleInteraction.py]: Quotes around the file path? Spaces are possible. Or better - quotes *if* spaces exist, no otherwise. Don't change the `file_path` property function though - we might need the raw filename later on.

Comment (adtac) [inline coalib/output/ConsoleInteraction.py]: Use `repr(bear.file_path)` if spaces exist so that you don't need to worry about single quotes, double quotes, ...

File: tests/output/ConsoleInteractionTest.py [hunk 5]
```diff
@@ -714,7 +718,9 @@ def test_show_bear_with_content(self):
                              "Optional, defaults to 'None'."
                              ')\n\n'
                              '  Can detect:\n   * Formatting\n\n'
-                             '  Can fix:\n   * Formatting\n\n')
+                             '  Can fix:\n   * Formatting\n\n'
+                             '  Path:\n'
+                             '   ' + repr(TestBear.file_path) + '\n\n\n')
```
Comment (adtac) [inline tests/output/ConsoleInteractionTest.py]: Why the change from two `\n\n` to three `\n\n\n` at the end? :(

I think only one `\n` at the end of the `console_printer.print` in `ConsoleInteraction.py`.

LGTM otherwise :)

**Comment by yash-nisar** [comment]
@adtac Is it okay ?

File: coalib/bears/Bear.py [hunk 6]
```diff
@@ -155,6 +156,13 @@ def can_detect(cls):
         return cls.CAN_DETECT | cls.CAN_FIX
 
     @classproperty
+    def file_path(cls):
```
Comment (sils) [inline coalib/bears/Bear.py]: this isn't unique enough, can we rathe name it `source_location` or so?

Comment (yash-nisar) [inline coalib/bears/Bear.py]: I'm on it ! Just a moment. ;)

File: coalib/bears/Bear.py [hunk 7]
```diff
@@ -155,6 +156,13 @@ def can_detect(cls):
...
+        """
```
Comment (sils) [inline coalib/bears/Bear.py]: can we have a proper description and a doctest like:

```
>>> class SomeBear(Bear): pass
>>> SomeBear.source_location
'...Bear.py'
```

**Comment by yash-nisar** [comment]
@sils @adtac @Makman2 Does this satisfy your requirements ?

File: coalib/bears/Bear.py [hunk 8]
```diff
@@ -121,6 +122,13 @@ class Bear(Printer, LogPrinterMixin):
     ...     BEAR_DEPS = {SomeBear}
     >>> SomeOtherBear.BEAR_DEPS
     {<class 'coalib.bears.Bear.SomeBear'>}
+
+    Every bear resides in some directory which is specified by the
+    source_location attribute:
+
+    >>> class SomeBear(Bear): pass
+    >>> SomeBear.source_location #doctest: +ELLIPSIS
```
Comment (Makman2) [inline coalib/bears/Bear.py]: we have `+ELLIPSIS` enabled by default :+1:

Comment (yash-nisar) [inline coalib/bears/Bear.py]: Okay, thanks 😄

Comment (yash-nisar) [inline coalib/bears/Bear.py]: Does the description seem good to you ?

Comment (Makman2) [inline coalib/bears/Bear.py]: which description?

Comment (yash-nisar) [inline coalib/bears/Bear.py]: > Every bear resides in some directory which is specified by the source_location attribute:

^The one that I added in the docstring.

**Comment by yash-nisar** [comment]
Mergeeeeeeeeeee !

**Comment by yash-nisar** [comment]
Mergeeeeeee !

**Comment by sils** [comment]
calm down! :D

https://github.com/coala/coala/pull/3559/commits/af800e05b3f250d4bad3b9a6134984fac4413682 is ready

**Comment by sils** [comment]
@rultor merge

**Comment by yash-nisar** [comment]
Hahaha @sils, was just fooling around. :laughing:
